app/subscriber/src/features/my-reports/edit/content/ContentEditForm.tsx

```diff
@@ -27,17 +30,67 @@ export interface IContentEditFormProps {
 export const ContentEditForm = React.forwardRef<HTMLDivElement | null, IContentEditFormProps>(
   ({ disabled }, ref) => {
     const [{ userInfo }] = useApp();
+    const [{ myTonePool, init }, { storeMyTonePool }] = useProfileStore();
     const [, { updateReport }] = useReports();
     const [, { addContent, updateContent }] = useContent();
+    const [, { addMyTonePool, getMyTonePool }] = useTonePool();
     const { values, onNavigate, isSubmitting, setSubmitting, setValues, activeRow, setActiveRow } =
       useReportEditContext();
 
     const [form, setForm] = React.useState<IReportInstanceContentForm | undefined>(activeRow);
     const [errors, setErrors] = React.useState<IContentValidationErrors>();
+    const [userFormTonePool, setUserFormTonePool] = React.useState<
+      IContentTonePoolModel | undefined
+    >(undefined);
 
     const instance = values.instances.length ? values.instances[0] : undefined;
     const userId = userInfo?.id ?? 0;
 
+    React.useEffect(() => {
+      const getTonePool = async () => {
+        try {
+          if (!init.myTonePool && userId !== 0) {
+            const response = await getMyTonePool(userId);
+            if (response?.id) {
+              storeMyTonePool(response);
+            } else {
+              // If no valid tone pool exists, proceed to create one
+              await createTonePool(userId);
+            }
+          }
+        } catch (error) {
+          console.error('Error loading tone pool:', error);
+        } finally {
+        }
+      };
+
+      const createTonePool = async (userId: number) => {
+        try {
+          await addMyTonePool({
+            ...defaultTonePool,
+            name: `${userId}`,
+            ownerId: userId,
+          });
+          const newTonePool = await getMyTonePool(userId);
```

Comment:
Extra request isn't needed, just use the returned tonepool from the `addMyTonePool`.